src/main/java/net/minecraftforge/fml/client/ClientHooks.java

```diff
@@ -76,108 +68,78 @@
     private static final CharMatcher DISALLOWED_CHAR_MATCHER = CharMatcher.anyOf(ALLOWED_CHARS).negate();
 
-    private static Map<ServerStatusResponse,JsonObject> extraServerListData;
-    private static Map<ServerData, ExtendedServerListData> serverDataTag;
-
-    public static void setupServerList()
+    private static final ResourceLocation iconSheet = new ResourceLocation(ForgeVersion.MOD_ID, "textures/gui/icons.png");
+    @Nullable
+    public static void processForgeListPingData(ServerStatusResponse packet, ServerData target)
     {
-        extraServerListData = Collections.synchronizedMap(new HashMap<>());
-        serverDataTag = Collections.synchronizedMap(new HashMap<>());
-    }
+        if(packet.getForgeData() != null){
+            int numberOfMods = packet.getForgeData().getNumberOfMods();
+            MapDifference<ResourceLocation, Integer> difference = Maps.difference(packet.getForgeData().getRegistryHashes(), RegistryManager.ACTIVE.computeRegistryHashes());
+            int fmlver = packet.getForgeData().getFMLNetworkVersion();
 
-    public static void captureAdditionalData(ServerStatusResponse serverstatusresponse, JsonObject jsonobject)
-    {
-        if (jsonobject.has("modinfo"))
-        {
-            JsonObject fmlData = jsonobject.get("modinfo").getAsJsonObject();
-            extraServerListData.put(serverstatusresponse, fmlData);
-        }
-    }
-    public static void bindServerListData(ServerData data, ServerStatusResponse originalResponse)
-    {
-/*
-        if (extraServerListData.containsKey(originalResponse))
-        {
-            JsonObject jsonData = extraServerListData.get(originalResponse);
-            String type = jsonData.get("type").getAsString();
-            JsonArray modDataArray = jsonData.get("modList").getAsJsonArray();
-            boolean moddedClientAllowed = !jsonData.has("clientModsAllowed") || jsonData.get("clientModsAllowed").getAsBoolean();
-            ImmutableMap.Builder<String, String> modListBldr = ImmutableMap.builder();
-            for (JsonElement obj : modDataArray)
-            {
-                JsonObject modObj = obj.getAsJsonObject();
-                modListBldr.put(modObj.get("modid").getAsString(), modObj.get("version").getAsString());
-            }
+            boolean b = NetworkRegistry.checkListPingCompatibilityForClient(packet.getForgeData().getPresentMods())
+                    && difference.areEqual()
```

Comment:
I believe it is intended to allow connections from clients with registries that are a superset of the registries on the server, so this check should be modified to support that.

Reply:
I'm aware that sending the complete registry during list ping is likely a lot of data, which is why I intended to send the hash of the set of all registry keys (Set#hashCode), which is an int *per registry*. The actual data send looks like this:
`{"description":{"text":"A Minecraft Server"},"players":{"max":20,"online":0},"version":{"name":"1.13.2","protocol":404},"forgeData":{"mods":[],"registryKeys":[{"namespace":"forge","path":"moddimensions","hash":0},{"namespace":"minecraft","path":"potiontypes","hash":196815442},{"namespace":"minecraft","path":"entities","hash":296289830},{"namespace":"minecraft","path":"biomes","hash":-587414690},{"namespace":"minecraft","path":"potions","hash":2094714475},{"namespace":"minecraft","path":"villagerprofessions","hash":-1709168725},{"namespace":"minecraft","path":"blocks","hash":-1813164851},{"namespace":"minecraft","path":"items","hash":413776779},{"namespace":"minecraft","path":"enchantments","hash":749676857},{"namespace":"minecraft","path":"tileentities","hash":-779590212},{"namespace":"minecraft","path":"soundevents","hash":-1298340753}],"numberOfMods":1,"fmlNetworkVersion":2}}`

What is an actual problem is that the client registry may indeed be a superset of the server registry, and this method does not allow for checking that.. I asked before implementing and was told the contents have to match, but apparently misinterpreted 'match' as 'be equal'..
So should I remove the gameregistry comparsion?